mesonbuild/modules/gnome.py

```diff
@@ -798,6 +799,14 @@ def generate_gir(self, state, args, kwargs):
             scan_command += ['--sources-top-dirs', os.path.join(state.environment.get_source_dir(), self.interpreter.subproject_dir, state.subproject)]
             scan_command += ['--sources-top-dirs', os.path.join(state.environment.get_build_dir(), self.interpreter.subproject_dir, state.subproject)]
 
+        if '--warn-error' in scan_command:
+            mlog.deprecation('Passing --warn-error is deprecated in favor of "fatal_warnings" keyword argument')
+        fatal_warnings = kwargs.get('fatal_warnings', False)
+        if not isinstance(fatal_warnings, bool):
+            raise MesonException('fatal_warnings keyword argument must be string.')
+        if fatal_warnings:
```

Comment:
Maybe this should only be done when enabled *and* `werror=true`? Otherwise it doesn't make sense, because it will _always_ be enabled, not just on the CI.

Reply:
Ah, I see, it leaves it up to the build file to add a custom option for it. Not sure I enjoy that, but ok.

Reply:
Yes, it's up to the user to decide, there was discussion there: https://github.com/mesonbuild/meson/issues/7130